src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py

```diff
@@ -340,3 +315,115 @@ def _resolve_api_version(client, provider_namespace, resource_type, parent_path)
+    elif database_engine == 'mysql':
+        connection_string = "Server={}; Port=3306; Database={}; Uid={}; Pwd={}; SslMode=Preferred;".format(server.fully_qualified_domain_name, database_name, administrator_login, administrator_login_password)
+        run_subprocess('gh secret set {} --repo {} -b"{}"'.format(AZURE_MYSQL_CONNECTION_STRING, repository, connection_string))
+
+
+def fill_action_template(cmd, database_engine, server, database_name, administrator_login, administrator_login_password, file_name, action_name, repository):
+
+    action_dir = get_git_root_dir() + GITHUB_ACTION_PATH
+    if not os.path.exists(action_dir):
+        os.makedirs(action_dir)
+
+    process = run_subprocess_get_output("gh secret list --repo {}".format(repository))
+    github_secrets = process.stdout.read().strip().decode('UTF-8')
+    connection_string = AZURE_POSTGRESQL_CONNECTION_STRING if database_engine == 'postgresql' else AZURE_MYSQL_CONNECTION_STRING
+    file_format = 'plsql-file' if database_engine == 'postgresql' else 'sql-file'
+
+    if AZURE_CREDENTIALS not in github_secrets:
+        register_credential_secrets(cmd,
+                                    server=server,
+                                    repository=repository)
+
+    if connection_string not in github_secrets:
+        register_connection_secrets(cmd,
+                                    database_engine=database_engine,
+                                    server=server,
+                                    database_name=database_name,
+                                    administrator_login=administrator_login,
+                                    administrator_login_password=administrator_login_password,
+                                    repository=repository)
+
+    condition = "on: [workflow_dispatch]\n"
+    yml_content = {
+        'jobs': {
+            'build': {
+                'runs-on': 'ubuntu-latest',
+                'steps': [
+                    {
+                        'uses': 'actions/checkout@v2.3.2'
+                    },
+                    {
+                        'uses': 'azure/login@v1',
+                        'with': {
+                            'creds': '${{secrets.AZURE_CREDENTIALS}}'
+                        }
+                    },
+                    {
+                        'uses': 'azure/{}@v1'.format(database_engine),
+                        'with': {
+                            'connection-string': '${{secrets.' + connection_string + '}}',
+                            'server-name': server.fully_qualified_domain_name,
+                            file_format: file_name,
+                        }
+                    }
+                ],
+            }
+        }
+    }
+
+    with open(action_dir + action_name + '.yml', 'w') as yml_file:
+        yml_file.write(condition)
+        yaml.dump(yml_content, yml_file)
```

Comment:
Can you create a real `template.yaml` with such content and just copy the file to `action_dir + action_name + '.yml'`? It's not that elegant to write all yaml content here...

Reply:
Fair point. Made the change.